app/helpers/schema_helper.py

```diff
@@ -129,23 +129,27 @@ def get_aliases(cls, survey_json):
                         raise QuestionnaireException('{} is not a unique alias'.format(answer['alias']))
         return aliases
 
-    @staticmethod
-    def get_answers_by_id_for_block(block_json):
+    @classmethod
+    def get_answers_by_id_for_block(cls, block_json):
         answers = {}
-        for section_json in block_json['sections']:
+        for section_json in cls._get_sections_in_block(block_json):
             for question_json in section_json['questions']:
                 for answer_json in question_json['answers']:
                     answers[answer_json['id']] = answer_json
         return answers
 
+    @staticmethod
```

Comment:
Not sure this adds much, also makes it a little inconsistent as you could say we should then do it for section_json['questions'] and question_json['answers'] as well